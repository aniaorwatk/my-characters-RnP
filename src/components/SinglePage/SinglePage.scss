@import "../../scss/Main";

.singlePageBox {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  line-height: 25px;
  margin: 10px;

  &__img {
    max-width: 50vh;
    max-height: 50vh;
    border-radius: 9px;
    box-shadow: 4px 4px 4px grey;
  }

  &__data {
    display: flex;
    align-items: center;
    flex-direction: column;

    &-textDnd {
      width: 80%;
      text-align: justify;
      margin: 10px 0;

      &--title {
        font-weight: bold;
      }
    }

    &-textCharacter {
      margin: 5px;
      font-size: $font-size-L;
      font-family: "Moon Dance", cursive;
    }

    &-titleHistory {
      font-size: $font-size-M;
      position: relative;
      margin-bottom: 30px;

      &::before {
        content: "";
        position: absolute;
        width: 30px;
        height: 4px;
        border-radius: 5px;
        background-color: gray;
        transform: translateX(50%);
        top: 30px;
      }
    }

    &-history {
      width: 80%;
      text-align: justify;
    }
  }
}

.singlePageBox > a {
  align-self: flex-end;
  margin-bottom: 20px;
}
