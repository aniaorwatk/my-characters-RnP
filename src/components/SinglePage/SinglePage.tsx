import React from "react";

const SinglePage = () => {};

export default SinglePage;
