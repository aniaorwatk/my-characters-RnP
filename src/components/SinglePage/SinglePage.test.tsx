import { render, screen, } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import SinglePage from './SinglePage';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useParams: () => ({
        id: 1,
    }),
}));

it('renders snapshot', () => {
    const tree = render(
        <MemoryRouter>
            <SinglePage />
        </MemoryRouter>
    )

    expect(tree.baseElement).toMatchSnapshot();
});