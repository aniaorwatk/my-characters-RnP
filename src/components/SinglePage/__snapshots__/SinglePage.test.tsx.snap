// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<body>
  <div>
    <div
      class="singlePageBox"
    >
      <img
        alt="MARGOT THORN"
        class="singlePageBox__img"
        src="https://images90.fotosik.pl/633/5c4bcae9f7462a49med.jpg"
      />
      <div
        class="singlePageBox__data"
      >
        <p
          class="singlePageBox__data-textCharacter"
        >
          MARGOT THORN
        </p>
        <div
          class="singlePageBox__data-textDnd"
        >
          <p>
            <span
              class="singlePageBox__data-textDnd--title"
            >
              Personality traits: 
            </span>
            Miła, pogodna duchem, zawsze uśmiechnięta, stara się we wszystkim widzieć same pozytywy
          </p>
          <p>
            <span
              class="singlePageBox__data-textDnd--title"
            >
              Ideals: 
            </span>
            Znaleźć sens życia, zostać zapamiętaną, pomagać innym
          </p>
          <p>
            <span
              class="singlePageBox__data-textDnd--title"
            >
              Bonds: 
            </span>
            Ojciec- Tymoteusz
          </p>
          <p>
            <span
              class="singlePageBox__data-textDnd--title"
            >
              Flaws: 
            </span>
            Jest powolna, szybko się męczy, trochę roztrzepana, stara się wszystkim pomagać co nie zawsze dobrze wychodzi lub pomaga nieodpowiednim ludziom
          </p>
        </div>
        <p
          class="singlePageBox__data-titleHistory"
        >
          History
        </p>
        <p
          class="singlePageBox__data-history"
        >
           Wywodzi się z bardzo szanowanej rodziny, która od pokoleń mieszka w Brackenmill. Jej rodzina od pokoleń zajmuje się magią, wywodzi się z niej wiele potężnych i szanowanych magów. Do których Margot niestety nie należała, mimo ogromnych starań i długoletniej nauki w Avium, jej umiejętności były przeciętne. Przez całe życie starała się pomagać innym i żyć w zgodzie z naukami, naturą. Lecz nigdy się nie wybiła, była po prostu tłem/ ścianą dla reszty rodziny. Odkąd była mała zawsze wierzył w nią jej ojciec. Który bardzo mocno ją kochał i mimo iż Margot nie była wspaniałym magiem na tle rodzeństwa i kuzynostwa, to ojciec zawsze się nią chwalił i o niej pamiętał, nie tak jak reszta rodziny. Zawsze wierzył, że ona czegoś dokona, że będzie szczęśliwa. Lata mijały a Margot nic nie osiągnęła. Bardzo ją to trapiło. Margot miała jeden dar, potrafiła przewidzieć przyszłość, co czasem jej pomogło w życiu. Pewnego ranka Margot obudziła się zlana cała potem, ponieważ zobaczyła swoją przyszłość. Niestety nie była ona różowa. Margot zobaczyła swoją śmierć w samotności, w tym samym łóżku w którym teraz leżała. Postanowiła wtedy, że musi coś zmienić, zacząć działać, inaczej zostanie całkiem sama. Jeszcze tego samego ranka spakowała się i pożegnała z ojcem. Powiedziała mu, że musi znaleźć jakiś cel w życiu, a nie tylko przemieszczać się jak cień po domu. Tak więc, Margot wyruszyła na swoją przygodę. Gdzieś na szlaku w stronę Alderheart, zamieszkała pod jednym z korzeni, tworząc tam wspaniały kolorowy domek, z ogródkiem, w którym zawsze pachnie ziołami i smacznymi warzywami. Nie mając większych perspektyw co może ze sobą zrobić i za co żyć, wpadła na pomysł, że użyje swoich zdolności. I tak oto zaczęła wróżyć i pomagać innym. Nigdy nikomu nie odmówiła pomocy, gdy przychodził ktoś do niej po poradę lub pomoc. Dla każdego znalazła się u niej zawsze herbatka i dobra rada. I tak oto Margot znalazła coś co sprawiało jej radość przez wiele, wiele, wiele lat. Nie miała już wizji swojej śmierci. Ale mimo to, żyje z nadzieją, że jest przeznaczony gdzieś na świecie jeszcze jakiś plan/cel dla niej.
        </p>
      </div>
    </div>
  </div>
</body>
`;
